common4j/src/main/com/microsoft/identity/common/java/controllers/BaseController.java

```diff
@@ -437,20 +448,26 @@ protected void renewAccessToken(@NonNull final SilentTokenCommandParameters para
                     methodTag,
                     "Token request was successful"
             );
-
-            // Suppressing unchecked warnings due to casting of rawtypes to generic types of OAuth2TokenCache's instance tokenCache while calling method saveAndLoadAggregatedAccountData
-            @SuppressWarnings(WarningType.unchecked_warning) final List<ICacheRecord> savedRecords = tokenCache.saveAndLoadAggregatedAccountData(
-                    strategy,
-                    getAuthorizationRequest(strategy, parameters),
-                    tokenResult.getTokenResponse()
-            );
-
-            final ICacheRecord savedRecord = savedRecords.get(0);
+            List<ICacheRecord> acquireTokenResultRecords = null;
+            if (parameters.hasNestedAppParameters()) {
+                // Do not save the token in naa flow
```

Comment:
Can you add more comments here indicating the rationale behind this?

Reply:
Taking this up in the next PR I am creating for adding flighting for NAA